### ISSUE TITLE
i found if we are using  rest props of es6 spread operator it will broken in below behaivor?

### ISSUE THREAD

**Issue by tianyingchun** [wmonk/create-react-app-typescript#64]
``` typescript

import * as React from 'react';
import Demo2 from './Demo2';

export default class XX extends React.Component<any, any> {

  render() {
    // console.log(Demo2);
    const { children, ...restProps } = this.props;
    // const x= {};
    console.log(restProps);
    return (
      <div>demo1 {this.props.name}
        <div {...restProps}>
          <Demo2></Demo2>
        </div>
      </div>
    )
  }
}
```
if we use above code block the vs.code will give me the Demo2 is declared but not used, while build process, it will lose <Demo2> is undefined.

if we change {...restProps}  to {...x}  it works fine. why?

**Comment by AJamesPhillips** [comment]
That doesn't make any sense, are you sure?  If it works with one unused variable it should work with another.

**Comment by DanielRosenwasser** [comment]
This is part of https://github.com/Microsoft/TypeScript/issues/15469 and is fixed in TypeScript 2.3.

**Comment by wmonk** [comment]
@tianyingchun please update to latest version of react scripts:

```npm install --save react-scripts@latest```
